src/main/java/haveric/recipeManager/RecipeManager.java

```diff
@@ -175,8 +176,15 @@ public void reload(CommandSender sender, boolean check, boolean firstTime) {
             }
 
             if (!firstTime && !Settings.getInstance().getClearRecipes()) {
-                Vanilla.restoreAllButSpecialRecipes();
-                Recipes.getInstance().index.putAll(Vanilla.initialRecipes);
+                if (!Version.has1_12Support()) {
+                    Vanilla.restoreAllButSpecialRecipes();
+                    Recipes.getInstance().index.putAll(Vanilla.initialRecipes);
+                } else {
+                    try {
+                        Bukkit.getServer().resetRecipes();
+                        Vanilla.init();
+                    } catch (NullPointerException npe) { /* Test Framework hates this */ }

```

Comment:
Actually, hadn't tried that specific spot without the try/catch. However, that same pattern elsewhere causes test failures as no mock was set for those methods (best guess, didn't look into it closely).